programs/bench.c

```diff
@@ -691,10 +699,13 @@ static BMK_return_t BMK_benchMemAdvancedNoAlloc(
     }
     DISPLAYLEVEL(2, "%2i#\n", cLevel);
     }   /* Bench */
+    results.result.cMem = (1ULL << (comprParams->windowLog)) + ZSTD_sizeof_CCtx(ctx);
+    results.error = 0;
     return results;
 }
 
 BMK_return_t BMK_benchMemAdvanced(const void* srcBuffer, size_t srcSize,
+                        void* dstBuffer, size_t dstCapacity, 
```

Comment:
according to function description,
if `dstBuffer==NULL`, then `dstCapacity==0`.
And I guess the reverse condition is also expected ?

`assert()` such condition.
Otherwise, you would have to deal with such case in the code flow,
and currently, a case like `dstBuffer==NULL` but `dstCapacity>0` doesn't end well.